modules/tinymce/src/plugins/codesample/main/ts/core/CodeSample.ts

```diff
@@ -27,21 +27,24 @@ const insertCodeSample = (editor: Editor, language: string, code: string) => {
 
     code = DOMUtils.DOM.encode(code);
 
-    return node.fold(() => {
+    if (Maybes.isNothing(node)) {
```

Comment:
Should this have used `Maybes.fold`? As that was why that was added I thought.

Reply:
I'm not gonna try and convince you that I've been 100% consistent here, but I would prefer to think of fold as a ternary equivalent. Not in the sense that it makes code convoluted and should only be used in simple cases, but more in the sense that it's meant to return something. As I didn't need to return something here, I didn't use it.

(Also, unless you've already bit the bullet with `Fun.pipe`, calling curried functions is painful, both for parenthesis soup and type annotation reasons)

Reply:
Could we have curried and uncurried versions. Sometimes you might just want to  call a single function going though a pipe just to avoid the uncurry seems a bit strange. Maybe have things like Maybes.mapU that is uncurried.